src/odemis/gui/cont/settings.py

```diff
@@ -719,3 +723,200 @@ def __init__(self, tab_panel, tab_data):
+        :parameter mode: (constant) SAVE, LOAD, ERROR or STATUS for display
+        :parameter filename: (str) filename of the loaded/saved file
+        """
+        if mode in (LOAD, SAVE):
+            self.panel.txt_StreakCalibFilename.Value = "%s" % filename
+            self.panel.txt_StreakCalibFilename.SetForegroundColour(odemis.gui.FG_COLOUR_EDIT)
+        elif mode == ERROR:
+            self.panel.txt_StreakCalibFilename.Value = "Error while loading file!"
+            self.panel.txt_StreakCalibFilename.SetForegroundColour(odemis.gui.FG_COLOUR_HIGHLIGHT)
+        elif mode == STATUS:
+            self.panel.txt_StreakCalibFilename.Value = "Calibration not saved yet!"
+            self.panel.txt_StreakCalibFilename.SetForegroundColour(odemis.gui.FG_COLOUR_WARNING)
+        else:
+            raise ValueError("Mode %s for display of the current status of the trigger delay "
+                             "calibration file is unknown.", mode)
+
+    def _onOpenCalibFile(self, event):
+        """
+        Loads a calibration file (*csv) containing the time range and the corresponding trigger delay
+        for streak camera calibration.
+        """
+        logging.debug("Open trigger delay calibration file for temporal acquisition.")
+
+        dialog = wx.FileDialog(self.panel,
+                               message="Choose a calibration file to load",
+                               defaultDir=self._calib_path,
+                               defaultFile="",
+                               style=wx.FD_OPEN,
+                               wildcard="csv files (*.csv)|*.csv")
+
+        # Show the dialog and check whether is was accepted or cancelled
+        if dialog.ShowModal() != wx.ID_OK:
+            return
+
+        # get selected path + filename and update default directory
+        self._calib_path = dialog.GetDirectory()
+        path = dialog.GetPath()
+        filename = dialog.GetFilename()
+
+        # read file
+        with open(path, 'rb') as csvfile:
+            calibFile = csv.reader(csvfile, delimiter=':')
+
+            try:
+                triggerDelay_dict = calibration.get_time_range_to_trigger_delay(calibFile,
+                                                                                self.streak_unit.timeRange.choices,
+                                                                                self.streak_delay.triggerDelay.range)
+            except ValueError as error:
+                self._onUpdateTriggerDelayGUI(ERROR, filename)  # update txt displayed in GUI
+                logging.error(error)
+                return
+
+        # update the MD
+        self.streak_delay.updateMetadata({model.MD_TIME_RANGE_TO_DELAY: triggerDelay_dict})
+
+        # update triggerDelay shown in GUI
+        cur_timeRange = self.streak_unit.timeRange.value
+        # find the corresponding trigger delay
+        tr2d = self.streak_delay.getMetadata()[model.MD_TIME_RANGE_TO_DELAY]
```

Comment:
That should be the same as `triggerDelay_dict`, as you've _just_ updated the metadata. => no need to read it again. BTW, I prefer `tr2d` name, not just because it's shorter, but also because it implies that timeRange information is also contained.